x-pack/plugins/index_management/server/routes/api/templates/register_create_route.ts

```diff
@@ -15,54 +15,44 @@ import { saveTemplate, doesTemplateExist } from './lib';
 
 const bodySchema = templateSchema;
 
-export function registerCreateRoute({ router, lib }: RouteDependencies) {
+export function registerCreateRoute({ router, lib: { handleEsError } }: RouteDependencies) {
   router.post(
     { path: addBasePath('/index_templates'), validate: { body: bodySchema } },
-    async (ctx, req, res) => {
-      const { callAsCurrentUser } = ctx.dataManagement!.client;
-      const template = req.body as TemplateDeserialized;
-      const {
-        _kbnMeta: { isLegacy },
-      } = template;
-
-      // Check that template with the same name doesn't already exist
-      const templateExists = await doesTemplateExist({
-        name: template.name,
-        callAsCurrentUser,
-        isLegacy,
-      });
-
-      if (templateExists) {
-        return res.conflict({
-          body: new Error(
-            i18n.translate('xpack.idxMgmt.createRoute.duplicateTemplateIdErrorMessage', {
-              defaultMessage: "There is already a template with name '{name}'.",
-              values: {
-                name: template.name,
-              },
-            })
-          ),
-        });
-      }
+    async (context, request, response) => {
+      const { client } = context.core.elasticsearch;
+      const template = request.body as TemplateDeserialized;
 
       try {
-        // Otherwise create new index template
-        const response = await saveTemplate({ template, callAsCurrentUser, isLegacy });
+        const {
+          _kbnMeta: { isLegacy },
+        } = template;
+
+        // Check that template with the same name doesn't already exist
+        const { body: templateExists } = await doesTemplateExist({
+          name: template.name,
+          client,
+          isLegacy,
+        });
 
-        return res.ok({ body: response });
-      } catch (e) {
-        if (lib.isEsError(e)) {
-          const error = lib.parseEsError(e.response);
-          return res.customError({
-            statusCode: e.statusCode,
-            body: {
-              message: error.message,
-              attributes: error,
-            },
+        if (templateExists) {
+          return response.conflict({
+            body: new Error(
+              i18n.translate('xpack.idxMgmt.createRoute.duplicateTemplateIdErrorMessage', {
+                defaultMessage: "There is already a template with name '{name}'.",
+                values: {
+                  name: template.name,
+                },
+              })
+            ),
           });
         }
-        // Case: default
-        throw e;
+
+        // Otherwise create new index template
+        const { body: responseBody } = await saveTemplate({ template, client, isLegacy });
+
+        return response.ok({ body: responseBody });
+      } catch (error) {
+        return handleEsError({ error, response });
```

Comment:
Does the `handleEsError` util provide the same functionality as before? I see previously it looks like we were parsing the error and passing it to `body.attributes`.

```
const error = lib.parseEsError(e.response);
```

Reply:
I just tested this. Looks like errors are surfaced correctly by the UI:

![image](https://user-images.githubusercontent.com/1238659/127722575-2b3942ee-c6ec-449e-a340-c15820817565.png)

![image](https://user-images.githubusercontent.com/1238659/127722579-40fa7f92-774d-4dde-9e48-eba603ce89e2.png)